linera-views/src/backends/scylla_db.rs

```diff
@@ -35,16 +31,59 @@ use crate::metering::MeteredStore;
 use crate::store::TestKeyValueStore;
 use crate::{
     batch::UnorderedBatch,
-    common::get_upper_bound_option,
+    common::{get_uleb128_size, get_upper_bound_option},
     journaling::{DirectWritableKeyValueStore, JournalConsistencyError, JournalingKeyValueStore},
     lru_caching::{LruCachingStore, LruSplittingConfig},
     store::{
         AdminKeyValueStore, CommonStoreInternalConfig, KeyValueStoreError, ReadableKeyValueStore,
         WithError,
     },
+    value_splitting::{ValueSplittingError, ValueSplittingStore},
 };
 
-/// The client for ScyllaDb.
+/// Fundamental constant in ScyllaDB: The maximum size of a multi keys query
+/// The limit is in reality 100. But we need one entry for the root key.
+const MAX_MULTI_KEYS: usize = 99;
+
+/// The maximal size of an operation on ScyllaDB seems to be 16M
+/// https://www.scylladb.com/2019/03/27/best-practices-for-scylla-applications/
+/// "There is a hard limit at 16MB, and nothing bigger than that can arrive at once
+///  at the database at any particular time"
+/// So, we set up the maximal size of 16M - 10K for the values and 10K for the keys
+/// We also arbitrarily decrease the size by 4000 bytes because an amount of size is
+/// taken internally by the database.
+const RAW_MAX_VALUE_SIZE: usize = 16762976;
+const MAX_KEY_SIZE: usize = 10240;
+const MAX_BATCH_TOTAL_SIZE: usize = RAW_MAX_VALUE_SIZE + MAX_KEY_SIZE;
+
+/// The `RAW_MAX_VALUE_SIZE` is the maximum size on the ScyllaDB storage.
+/// However, the value being written can also be the serialization of a SimpleUnorderedBatch
+/// Therefore the actual `MAX_VALUE_SIZE` is lower.
+/// At the maximum the key_size is 1024 bytes (see below) and we pack just one entry.
+/// So if the key has 1024 bytes this gets us the inequality
+/// `1 + 1 + 1 + serialized_size(MAX_KEY_SIZE)? + serialized_size(x)? <= RAW_MAX_VALUE_SIZE`.
+/// and so this simplifies to `1 + 1 + 1 + (2 + 10240) + (4 + x) <= RAW_MAX_VALUE_SIZE`
+/// Note on the above formula:
+/// * We write 4 because `get_uleb128_size(RAW_MAX_VALUE_SIZE) = 4)`
+/// * We write `1 + 1 + 1`  because the `UnorderedBatch` has three entries.
+///
+/// This gets us to a maximal value of 16752727.
+const VISIBLE_MAX_VALUE_SIZE: usize = RAW_MAX_VALUE_SIZE
+    - MAX_KEY_SIZE
+    - get_uleb128_size(RAW_MAX_VALUE_SIZE)
+    - get_uleb128_size(MAX_KEY_SIZE)
+    - 1
+    - 1
+    - 1;
+
+/// The constant 14000 is an empirical constant that was found to be necessary
+/// to make the ScyllaDB system work. We have not been able to find this or
+/// a similar constant in the source code or the documentation.
+/// An experimental approach gets us that 14796 is the latest value that is
+/// correct.
+const MAX_BATCH_SIZE: usize = 5000;
```

Comment:
The numbers in the comment still don't match the value.

Reply:
@MathieuDutSik: Sorry I saw this too late. Maybe it can be addressed in a follow-up PR?